src/compute-engine/library/arithmetic.ts

```diff
@@ -184,9 +183,9 @@ export const ARITHMETIC_LIBRARY: IdentifierDefinitions[] = [
         if (ops.some((x) => x.isReal === false || x.isNaN)) return 'unsigned';
         if (ops.every((x) => x.is(0))) return 'zero';
         if (ops.every((x) => x.isNonNegative))
-          return ops.every((x) => x.isPositive) ? 'positive' : 'non-negative';
+          return ops.some((x) => x.isPositive) ? 'positive' : 'non-negative';
         if (ops.every((x) => x.isNonPositive))
-          return ops.every((x) => x.isNegative) ? 'negative' : 'non-positive';
+          return ops.some((x) => x.isNegative) ? 'negative' : 'non-positive';
         if (ops.every((x) => x.isReal)) return 'real';
```

Comment:
I don't think this can ever be triggered, since there is a check for isReal === false at the beginning of the handler.

Reply:
You're right but your top line is wrong. If some x is not real it should return undefined not unsigned (consider i+(-i)) so you need to remove that. So you would do a test for x.isNaN to return unsigned. This means that the test for isReal is no longer there which is why the other line is needed.

Reply:
I don't think it's "my" top line, but if it's incorrect it should be fixed.